src/Core/AdminConsole/OrganizationFeatures/Policies/Enforcement/AutoConfirm/AutomaticUserConfirmationPolicyEnforcementValidator.cs

```diff
@@ -26,14 +26,17 @@ public async Task<ValidationResult<AutomaticUserConfirmationPolicyEnforcementReq
             return Invalid(request, new CurrentOrganizationUserIsNotPresentInRequest());
         }
 
-        if ((await providerUserRepository.GetManyByUserAsync(request.User.Id)).Count != 0)
+        if (automaticUserConfirmationPolicyRequirement.IsEnabled(request.OrganizationId))
         {
-            return Invalid(request, new ProviderUsersCannotJoin());
-        }
-
-        if (request.AllOrganizationUsers.Count == 1)
-        {
-            return Valid(request);
+            if ((await providerUserRepository.GetManyByUserAsync(request.User.Id)).Count != 0)
+            {
+                return Invalid(request, new ProviderUsersCannotJoin());
+            }
+
+            if (request.AllOrganizationUsers.Count == 1)
+            {
+                return Valid(request);
+            }
```

Comment:
This needs to be inverted, otherwise if `AllOrganizationUsers.Count > 1`, it will fall through to returning `Valid` at the end of the method. It's also generally clearer to check the prohibited cases.

```c#
            if (request.AllOrganizationUsers.Any(ou => ou !== currentOrganizationUser))
            {
                return Invalid(request, new WhateverYourErrorMessageIs());
            }
```